app/javascript/components/generic-objects-form/index.jsx

```diff
@@ -0,0 +1,234 @@
+        return __(`File is too large, maximum allowed size is ${maxSize} bytes. Current file has ${value.inputFiles[0].size} bytes`);
+    };
+  };
+
+  //custom validator that makes sure attribute, association, and method names can only contain lowercase letters, numbers or underscores
+  const syntaxValidator = () => {
+    var format = /[ `!@#$%^&*()+\-=\[\]{};':"\\|,.<>\/?~]/;
+    return (value) => {
+      if (value != value.toLowerCase() || format.test(value))
+        return __(`Name can only contain lowercase letters, numbers, or underscores`);
+    };
+  };
+
+  //custom validator that checks to make sure no two attributes/associations/methods have the same name
+  const uniqueNameValidator = (values) => {
+    const errors = { attributes: [], associations: [], methods: [] };
+
+    for (const index in values.attributes) {
+      for (const position in values.attributes) {
+        if (values.attributes[index] != undefined && values.attributes[position] != undefined) {
+          if (index != position && values.attributes[index].attributes_name == values.attributes[position].attributes_name)
+            errors.attributes[position] = { attributes_name: __('Name must be unique') };
+        }
+      }
+    }
+
+    for (const index in values.associations) {
+      for (const position in values.associations) {
+        if (values.associations[index] != undefined && values.associations[position] != undefined) {
+          if (index != position && values.associations[index].associations_name == values.associations[position].associations_name)
+            errors.associations[position] = { associations_name: __('Name must be unique') };
+        }
+      }
+    }
+
+    for (const index in values.methods) {
+      for (const position in values.methods) {
+        if (values.methods[index] != undefined && values.methods[position] != undefined) {
+          if (index != position && values.methods[index].methods_name == values.methods[position].methods_name)
+            errors.methods[position] = { methods_name: __('Name must be unique') };
+        }
+      }
+    }
+
+    return errors;
+  };
+
+  //custom component and validator mappers that combine the above components/validators with the existing mappers used in manageiq
+  const mapper = {
+    ...componentMapper,
+    'file-edit': FileEditComponent,
+  };
+
+  const validMapper = {
+    'file': fileValidator,
+    'syntax': syntaxValidator,
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      miqSparkleOn();
+      API.get(`/api/generic_object_definitions/${recordId}?attributes=picture.image_href`).then(initialValues => {
+        promise.then(({ data: { allowed_association_types } }) => {
+          initialValues.attributes = [];
+          initialValues.associations = [];
+          initialValues.methods = [];
+
+          //modifies the attributes/associations/methods data from the API to match what the form is expecting
```

Comment:
for.. in loops giving lint errors as they are not recommended, there are other options, check https://stackoverflow.com/a/5737136/8808023